### PR TITLE
Remove all runtime dependencies

### PR DESCRIPTION
**Pull request by dcylabs**
As I inspected the polyglot code in [this issue](https://github.com/airbnb/polyglot.js/pull/149) and saw that some dependencies are not maintained since 2 years ago. I wanted to do a little of **boy scout attitude** and propose a PR with the removal all dependencies to a simplistic way, so it doesnt need any runtime dependency anymore.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,13 +17,31 @@
...
+  var keys = Object.keys(iterable);
+  for (var i = 0; i < keys.length; i += 1) {
+    var key = keys[i];
+    var val = iterable[key];
+    callback(val, key, iterable);
+  }
+};
+
+var has = function (object, property) {
+  return Object.prototype.hasOwnProperty.call(object, property);
```
Comment (ljharb) [inline index.js:33]: this exposes the package to breakage from `delete Object.prototype.hasOwnProperty`, for example, that the `has` package protects against.

File: index.js [hunk 2]
```diff
@@ -17,13 +17,31 @@
...
+};
+
+var trim = function (input) {
+  return input.replace(/(^\s*)|(\s*$)/mg, '');
```
Comment (ljharb) [inline index.js:37]: this makes the package vulnerable to someone messing with `String.prototype.replace`, and doesn't properly handle whitespace (in the unicode sense) the way the trim package does.

File: index.js [hunk 3]
```diff
@@ -17,13 +17,31 @@
...
 
-var forEach = require('for-each');
-var warning = require('warning');
-var has = require('has');
-var trim = require('string.prototype.trim');
+/* Should always be false */
+var debug = false;
+
+var forEach = function (iterable, callback) {
+  var keys = Object.keys(iterable);
```
Comment (ljharb) [inline index.js:24]: this breaks in engines that don't have `Object.keys`, and also exposes the package to breakage from `delete Object.keys`, eg.

**Comment by dcylabs** [comment]
I understand, closing the PR :)

File: index.js (hunk 2 above)
Comment (simon04) [reply index.js:37]: > this makes the package vulnerable to someone messing with `String.prototype.replace`

Who would do that? When I want to kill my JavaScript environment by randomly overriding prototype functions, polyglot.js won't save me…

**Comment by simon04** [comment]
Just recently, I wanted to use this library without a build system, simply by including https://unpkg.com/node-polyglot. Due to the external (mini) dependencies and not providing a browser-compatible bundle, I could not.

Please reconsider removing the almost trivial external dependencies in favour of a fully self-contained library!

**Comment by ljharb** [comment]
Modern web dev isn’t possible without using a build system regardless. This is a node module, and it’s expected node’s ecosystem’s conventions will be followed.

unpkg is not intended to let you bypass a build system; you may be thinking of https://wzrd.in

**Pull request closed without merging**